docs/troubleshooting.rst

```diff
@@ -0,0 +1,140 @@
+:Solution: The easiest way to solve this issue is to re-pack the experiment: write a simple bash script that runs the experiment and either renames outputs or creates symbolic links to them with known filenames; then, trace this script (instead of the actual entry-point of your experiment) and specify these fixed path names in the ``inputs_outputs`` section of the configuration file.
+
+------------
+
+..  _pycrypto_windows:
+
+:Issue: **"** `reprounzip-vagrant` **installation fails with error** ``Unable to find vcvarsall.bat`` **on Windows."**
+:Diagnosis: Python is trying to build `PyCrypto <https://www.dlitz.net/software/pycrypto/>`__, one of the dependencies of `reprounzip-vagrant`, but there is no C compiler available.
+:Solution: You can either build PyCrypto from source, or follow the instructions on `this website <http://stackoverflow.com/questions/11405549/how-do-i-install-pycrypto-on-windows>`__ to get the non-official binaries.
+
+------------
+
+..  _compiler_mac:
+
+:Issue: **"** `reprounzip-vagrant` **installation fails with error** ``unknown argument: '-mno-fused-madd'`` **on Mac OS X."**
+:Diagnosis: This is an issue with the Apple LLVM compiler, which treats unrecognized command-line options as errors.
+:Solution: As a workaround, before installing `reprounzip-vagrant`, run the following::
+
+               $ sudo -s
+               $ export CFLAGS="-Wno-error=unused-command-line-argument-hard-error-in-future"
+
+           Then, re-install `reprounzip-vagrant`::
+
+               $ pip install -I reprounzip-vagrant
+
+           Or use the following command in case you want all the available plugins::
+
+               $ pip install -I reprounzip[all]
+
+------------
+
+:Issue: **"The experiment fails with** ``Error: Can't open display: :0`` **when trying to reproduce it."**
+:Diagnosis: The experiment probably involves running a GUI tool.
+:Solution: The `reprounzip` component supports GUI tools, but it is not enabled by default; add the flag ``--enable-x11`` to the ``run`` command to enable it. See :ref:`gui-tools` for more information.
+
+------------
+
+..  _directory_error:
+
+:Issue: **"** `reprounzip directory` **fails with** ``IOError`` **."**
+:Diagnosis: The `directory` unpacker does not provide any isolation from the filesystem: if the experiment being reproduced use absolute paths, these will point outside the experiment directory, and files may not be found.
+:Solution: Make sure that the experiment does not use any absolute paths: if only relative paths are used internally and in the command line, ``reprounzip directory`` should work. As an alternative, you can use other unpackers (e.g.: ``reprounzip chroot`` and ``reprounzip vagrant``) that work in the presence of hardcoded absolute paths.
+
```

Comment:
I think you added this a long time ago, no?

I did not change any of the FAQ original questions / answers.

On Thu, Sep 17, 2015 at 2:15 PM, Remi Rampin <notifications@github.com>
wrote:

> In docs/troubleshooting.rst
> <https://github.com/ViDA-NYU/reprozip/pull/147#discussion_r39779730>:
>
> > +               $ pip install -I reprounzip[all]
> > +
> > +------------
> > +
> > +:Issue: **"The experiment fails with** ``Error: Can't open display: :0`` **when trying to reproduce it."**
> > +:Diagnosis: The experiment probably involves running a GUI tool.
> > +:Solution: The `reprounzip` component supports GUI tools, but it is not enabled by default; add the flag ``--enable-x11`` to the ``run`` command to enable it. See :ref:`gui-tools` for more information.
> > +
> > +------------
> > +
> > +..  _directory_error:
> > +
> > +:Issue: **"** `reprounzip directory` **fails with** ``IOError`` **."**
> > +:Diagnosis: The `directory` unpacker does not provide any isolation from the filesystem: if the experiment being reproduced use absolute paths, these will point outside the experiment directory, and files may not be found.
> > +:Solution: Make sure that the experiment does not use any absolute paths: if only relative paths are used internally and in the command line, ``reprounzip directory`` should work. As an alternative, you can use other unpackers (e.g.: ``reprounzip chroot`` and ``reprounzip vagrant``) that work in the presence of hardcoded absolute paths.
> > +
>
> Hmm, this doesn't seem right, why IOError? This will only be raised if
> the experiment is Python and doesn't handle it...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ViDA-NYU/reprozip/pull/147/files#r39779730>.
>



-- 
Fernando Chirigati
Research Assistant and Doctoral Candidate at NYU
http://bigdata.poly.edu/~fchirigati/ <http://vgc.poly.edu/~fchirigati/>